src/freertos_drivers/ti/CC32xxDeviceFile.cxx

```diff
@@ -92,13 +92,16 @@ int CC32xxDeviceFile::open(File* file, const char *path, int flags, int mode)
     else
     {
         /* file not open yet, open and intialize metadata */
-        int32_t result;
+        int32_t result = 0;
         if (flags & O_CREAT)
         {
             result = sl_FsOpen((const unsigned char *)path,
                                SL_FS_CREATE |
                                SL_FS_CREATE_MAX_SIZE(maxSizeOnCreate),
                                nullptr);
+        }
+        if (result > 0)
```

Comment:
this looks like it will break the original if/else if setup. Perhaps moving this block into the `O_CREAT` block? It also looks like `O_WRONLY` may also assume that write should be enabled.

Reply:
Actually, I think this logic is correct.

Reply:
Technically the flow is correct but it changes the original behavior of the if/else if on read/write vs write only. Write only also has the same bug of not checking the result code form opening the file and will assume it can write even if the file is not opened. Likely the solution would be move the result check further down and only set write enabled if the file was successfully opened for write mode (read or not)